### PR TITLE
Refs #34900 -- Updated requirements for Python 3.13.

### PR DESCRIPTION
**Pull request by felixxm**
ticket-34900

[Seems to work on Windows](https://github.com/felixxm/django/actions/runs/10505645512/job/29103609976).

### CODE REVIEW

File: tests/requirements/py3.txt [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
+argon2-cffi >= 19.2.0
 bcrypt
 black
 docutils >= 0.19
-geoip2; python_version < '3.13'
+geoip2
 jinja2 >= 2.11.0
-numpy; sys_platform != 'win32' or python_version < '3.13'
-Pillow >= 6.2.1; sys_platform != 'win32' or python_version < '3.13'
+numpy; python_version < '3.13'
```
Comment (felixxm) [inline tests/requirements/py3.txt:9]: `numpy` cannot be installed on Python 3.13 even on Linux:
```
...
      In file included from ../numpy/_core/src/multiarray/common.h:12,
                       from ../numpy/_core/src/multiarray/arraytypes.h.src:4,
                       from ../numpy/_core/src/multiarray/argfunc.dispatch.c.src:15:
      ../numpy/_core/src/common/npy_import.h:17:5: error: unknown type name ‘PyMutex’
         17 |     PyMutex import_mutex;
            |     ^~~~~~~
      ../numpy/_core/src/common/npy_import.h: In function ‘npy_cache_import_runtime’:
      ../numpy/_core/src/common/npy_import.h:96:9: warning: implicit declaration of function ‘PyMutex_Lock’ [-Wimplicit-function-declaration]
         96 |         PyMutex_Lock(&npy_runtime_imports.import_mutex);
            |         ^~~~~~~~~~~~
      ../numpy/_core/src/common/npy_import.h:104:9: warning: implicit declaration of function ‘PyMutex_Unlock’ [-Wimplicit-function-declaration]
        104 |         PyMutex_Unlock(&npy_runtime_imports.import_mutex);
            |         ^~~~~~~~~~~~~~
      [68/592] Compiling C object numpy/_core/_struct_ufunc_tests.cpython-314-x86_64-linux-gnu.so.p/src_umath__struct_ufunc_tests.c.o
      [69/592] Compiling C object numpy/_core/_umath_tests.cpython-314-x86_64-linux-gnu.so.p/src_common_npy_cpu_features.c.o
      [70/592] Compiling C object numpy/_core/lib_umath_tests.dispatch.h_AVX2.a.p/src_umath__umath_tests.dispatch.c.o
      [71/592] Compiling C++ object numpy/_core/libnpymath.a.p/src_npymath_halffloat.cpp.o
      [72/592] Compiling C object numpy/_core/libargfunc.dispatch.h_AVX512_SKX.a.p/meson-generated_argfunc.dispatch.c.o
      FAILED: numpy/_core/libargfunc.dispatch.h_AVX512_SKX.a.p/meson-generated_argfunc.dispatch.c.o
      cc -Inumpy/_core/libargfunc.dispatch.h_AVX512_SKX.a.p -Inumpy/_core -I../numpy/_core -Inumpy/_core/include -I../numpy/_core/include -I../numpy/_core/src/common -I../numpy/_core/src/multiarray -I../numpy/_core/src/multiarray/stringdtype -I../numpy/_core/src/npymath -I../numpy/_core/src/umath -I/home/felixx/.pyenv/versions/3.13-dev/include/python3.14 -I/tmp/pip-install-4d9khj38/numpy_e16cc86642d24f85ac06fc6848787e29/.mesonpy-t0c9vg1e/meson_cpu -fdiagnostics-color=always -DNDEBUG -Wall -Winvalid-pch -std=c11 -O3 -fno-strict-aliasing -msse -msse2 -msse3 -fPIC -DNPY_INTERNAL_BUILD -DHAVE_NPY_CONFIG_H -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE=1 -D_LARGEFILE64_SOURCE=1 -O3 -DNPY_HAVE_SSE2 -DNPY_HAVE_SSE -DNPY_HAVE_SSE3 -DNPY_HAVE_SSSE3 -DNPY_HAVE_SSE41 -DNPY_HAVE_POPCNT -DNPY_HAVE_SSE42 -DNPY_HAVE_AVX -DNPY_HAVE_F16C -DNPY_HAVE_FMA3 -DNPY_HAVE_AVX2 -DNPY_HAVE_AVX512F -DNPY_HAVE_AVX512F_REDUCE -DNPY_HAVE_AVX512CD -DNPY_HAVE_AVX512_SKX -DNPY_HAVE_AVX512VL -DNPY_HAVE_AVX512BW -DNPY_HAVE_AVX512DQ -DNPY_HAVE_AVX512BW_MASK -DNPY_HAVE_AVX512DQ_MASK -msse -msse2 -msse3 -mssse3 -msse4.1 -mpopcnt -msse4.2 -mavx -mf16c -mfma -mavx2 -maes -mpclmul -mbmi -mbmi2 -mno-mmx -mavx512f -mavx512cd -mavx512vl -mavx512bw -mavx512dq -DNPY_MTARGETS_CURRENT=AVX512_SKX -MD -MQ numpy/_core/libargfunc.dispatch.h_AVX512_SKX.a.p/meson-generated_argfunc.dispatch.c.o -MF numpy/_core/libargfunc.dispatch.h_AVX512_SKX.a.p/meson-generated_argfunc.dispatch.c.o.d -o numpy/_core/libargfunc.dispatch.h_AVX512_SKX.a.p/meson-generated_argfunc.dispatch.c.o -c numpy/_core/libargfunc.dispatch.h_AVX512_SKX.a.p/argfunc.dispatch.c
      In file included from ../numpy/_core/src/multiarray/common.h:12,
                       from ../numpy/_core/src/multiarray/arraytypes.h.src:4,
                       from ../numpy/_core/src/multiarray/argfunc.dispatch.c.src:15:
      ../numpy/_core/src/common/npy_import.h:17:5: error: unknown type name ‘PyMutex’
         17 |     PyMutex import_mutex;
            |     ^~~~~~~
      ../numpy/_core/src/common/npy_import.h: In function ‘npy_cache_import_runtime’:
      ../numpy/_core/src/common/npy_import.h:96:9: warning: implicit declaration of function ‘PyMutex_Lock’ [-Wimplicit-function-declaration]
         96 |         PyMutex_Lock(&npy_runtime_imports.import_mutex);
            |         ^~~~~~~~~~~~
      ../numpy/_core/src/common/npy_import.h:104:9: warning: implicit declaration of function ‘PyMutex_Unlock’ [-Wimplicit-function-declaration]
        104 |         PyMutex_Unlock(&npy_runtime_imports.import_mutex);
            |         ^~~~~~~~~~~~~~
      [73/592] Compiling C object numpy/_core/_multiarray_tests.cpython-314-x86_64-linux-gnu.so.p/src_common_mem_overlap.c.o
      ninja: build stopped: subcommand failed.
      [end of output]
```

Comment (felixxm) [reply tests/requirements/py3.txt]: Let's recheck this :thinking:

Comment (felixxm) [reply tests/requirements/py3.txt]: OK, it works, my bad.

**Review by pauloxnet — APPROVED** [review]

**Review by mahfuztbt — APPROVED** [review]

**Review by LilyFoote — APPROVED** [review]

**Review by nessita — APPROVED** [review]
Thank you for this work! :star2:

**Pull request merged**